### PR TITLE
Added Host[Device Name] To !fetch

### PR DESCRIPTION
**Pull request by ghost**
So that people can set whatever device they are using

### CODE REVIEW

**Comment by eepykate** [comment]
1.

image

2. This will just spit ~~an error~~ 2 errors on non-linux systems, right in the middle of the fetch output.

**Comment by ghost** [comment]
i think i can fix that

**Comment by Tanish2002** [comment]
As you can see from the build output you didn't change db/sysinfo.go to support `Host` 
look at this pull request for reference
https://github.com/unixporn/trup/pull/143/files

**Comment by eepykate** [comment]
Damn they write some nice code, can i get in touch?

**Comment by ghost** [comment]
Fixed everything :)

File: db/sysinfo.go [hunk 1]
```diff
@@ -91,15 +92,16 @@ func TopSysinfoFields() ([]TopSysinfo, error) {
...
-		UNION SELECT 3 as order, * FROM top_field('DisplayProtocol')
-		UNION SELECT 4 as order, * FROM top_field('Shell')
-		UNION SELECT 5 as order, * FROM top_field('Terminal')
-		UNION SELECT 6 as order, * FROM top_field('Bar')
-		UNION SELECT 7 as order, * FROM top_field('Gtk3Theme')
-		UNION SELECT 8 as order, * FROM top_field('GtkIconTheme')
-		UNION SELECT 9 as order, * FROM top_field('Editor')
-		UNION SELECT 10 as order, * FROM top_field('Cpu')
+	    UNION SELECT 2 as order, * FROM top_field('Host')
+        UNION SELECT 3 as order, * FROM top_field('DeWm')
```
Comment (eepykate) [inline db/sysinfo.go:96]: Looks like you broke the indenting on these lines

File: fetcher.sh [hunk 2]
```diff
@@ -12,6 +12,7 @@ Also note that !git, !dotfiles, and !desc are different commands.
 
 !setfetch
 Distro: $NAME $ver
+Host: $(cat /sys/devices/virtual/dmi/id/product_name)
```
Comment (eepykate) [inline fetcher.sh:15]: Please do not directly `cat` the file here, please put it in the linux section such as

```
read -r host <  /sys/devices/virtual/dmi/id/product_name
```
then use `Host: $host` here.

**Review by eepykate — CHANGES_REQUESTED** [review]

**Comment by 6gk** [comment]
Consider using something like this for getting the host.

```sh
# it's a long dir...
v=/sys/devices/virtual/dmi/id
# initial check, if this is set it *should* have more info
read -r host < $v/product_name
# if product_name is unset (ex: on a desktop), check for the motherboard
[ "$host" = "Default String" ] && read -r host < $v/board_name
# if it's all unset, reset the $host variable (since it'll just display useless information if kept) 
[ "$host" = "Default String" ] && host=
```

**Comment by eepykate** [comment]
Sounds like there's no easy way about cleanly detecting the model.

```
# example format
product_name | product_version

# smokin   (laptop)
## HP made their own placeholder, not sure how many others have done this, 
##  sounds like there will be a million edge cases
HP Pavilion Notebook | Type1ProductConfigId
# tanish   (laptop)
## order is wack
82EY | IdeaPad Gaming3 15ARH05D
# me   (desktop)
##  "Default string" is easy enough to remove since it seems to be global
H110M-H | Default string
# ladle   (desktop)
## seems reasonable enough
MS-7C02 | 1.0
```

**Comment by SmokinSpectre** [comment]
Won't work for most devices...so closing it

File: db/sysinfo.go (hunk 1 above)
Comment (ghost) [reply db/sysinfo.go:96]: oof yes :(

**Comment by SmokinSpectre** [comment]
Opening it again....people can find some good solution to this

**Pull request closed without merging**